udata/core/dataset/api.py

```diff
@@ -362,7 +363,10 @@ def handle_upload(self, dataset):
         if 'html' in infos['mime']:
             api.abort(415, 'Incorrect file content type: HTML')
         infos['title'] = os.path.basename(infos['filename'])
-        infos['checksum'] = Checksum(type='sha1', value=infos.pop('sha1'))
+        for checksum_type in CHECKSUM_TYPES:
+            if checksum_type in infos:
+                infos['checksum'] = Checksum(type=checksum_type, value=infos.pop(checksum_type))
+                break
```

Comment:
`infos['checksum']` may never be set if none of the checksum_type matches and it wouldn't raise an error here.
Using `next()` would raise an error directly instead, what do you think?

```suggestion
        checksum_type = next(algo for algo in CHECKSUM_TYPES if algo in infos)
        infos['checksum'] = Checksum(type=checksum_type, value=infos.pop(checksum_type))
```

Reply:
I was actually hesitating whether we want to raise an exception here or wait for the caller to complain, but after all I agree it's best to do it here. Also, looks way more elegant with `next()`.